library_samples/Python3/ocs_sample_library_preview/Authentication.py

```diff
@@ -0,0 +1,148 @@
+
+from .SdsError import SdsError
+
+import base64
+import configparser
+import hashlib
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import json
+import os 
+import secrets
+import time
+from urllib.parse import urlparse, parse_qs
+import webbrowser
+
+import requests
+
+
+class Authentication(object):
+
+    def __init__(self, tenant, url, clientId, clientSecret):
+        self.__tenant = tenant
+        self.__clientId = clientId
+        self.__clientSecret = clientSecret
+        self.__url = url
+
+        self.__expiration = 0
+        self.__token = ""
+        if clientSecret is not None:
+            self.__getToken = self.__getClientIDSecretToken
+        else:
+            self.__getToken = self.__getPKCEToken
+        
+
+    def getToken(self):
+        return self.__getToken()
+
+
+    def __getClientIDSecretToken(self):
+           
+        if ((self.__expiration - time.time()) > 5 * 60):
```

Comment:
So this means if the token is going to expire within the next 5 minutes, we go ahead and refresh / get a new token? Just making sure I understand. 
Since this check is shared with the other method it could be pulled out into its own function, or just moved inside the `getToken` method before calling `self.__getToken()`.

Reply:
Correct. Good suggestion, I'll just do it.

Reply:
done